### ISSUE TITLE
[css-transforms] `transform-box:stroke-box` with `vector-effect:non-scaling-stroke` creates a cyclical dependency

### ISSUE THREAD

**Issue by jwatt** [w3c/csswg-drafts#9640]
A cyclical dependency can be created if [`transform-box:stroke`](https://www.w3.org/TR/css-transforms-1/#transform-box) and [`vector-effect:non-scaling-stroke`](https://svgwg.org/svg2-draft/coords.html#VectorEffects) are both used on the same stroked SVG element. (And by implication, `transform-box:border-box` too, since the spec says "For SVG elements ... the used value for `border-box` is `stroke-box`.")

To calculate the stroke bounds of an element with `vector-effect:non-scaling-stroke` we need to resolve the transform to the element's outer-`<svg>`, but to resolve that transform when the element has `transform-box:stroke-box` requires the element's stroke bounds if the transform contains percentage values.

I don't think there's an ideal way to break this cyclical dependency, but I would suggest breaking it by requiring implementations to use `fill-box` in place of `stroke-box` (or `border-box`) if the element has `vector-effect:non-scaling-stroke`. (Probably regardless of whether it has a transform that contains percentage values, just to make implementation reasonable.)

**Comment by BorisChiou** [comment]
Yes. This may have impact on any CSS property which uses `stroke-box` as the reference box with `vector-effect:non-scaling-stroke` (e.g. https://drafts.fxtf.org/motion-1/#offset-path-property).

**Comment by svgeesus** [comment]
>  I would suggest breaking it by requiring implementations to use `fill-box` in place of `stroke-box` (or `border-box`) if the element has `vector-effect:non-scaling-stroke`. (Probably regardless of whether it has a transform that contains percentage values, just to make implementation reasonable.)

That solution (with `fill-box`) seems like a reasonable option to me. Most actual uses of non-scaling stroke are "a thin stroke, not so small as to be invisible, and not all chonky when you zoom in" so the difference between `stroke-box` and `fill-box` would typically only be a couple of `px` anyway.
